### PR TITLE
Remove bootstrap.proj

### PR DESCRIPTION
**Pull request by jeffkl**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2186

Regression? Last working version:

## Description

This removes `bootstrap.proj` and replaces it with more retail MSBuild logic and allows a user to clone our repository and build it as long as they have Visual Studio and .NET SDK installed.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -157,10 +157,15 @@ Invoke-BuildStep $VSMessage {
 -ev +BuildErrors
 
 Invoke-BuildStep 'Creating the EndToEnd test package' {
-        param($Configuration)
-        $EndToEndScript = Join-Path $PSScriptRoot scripts\cibuild\CreateEndToEndTestPackage.ps1 -Resolve
-        $OutDir = Join-Path $Artifacts VS15
-        & $EndToEndScript -c $Configuration -out $OutDir
+        $msbuildArgs = "test\TestUtilities\CreateEndToEndTestPackage\CreateEndToEndTestPackage.proj", "/p:Configuration=$Configuration", "/restore:false", "/property:BuildProjectReferences=false"
```
Comment (zivkan) [inline build.ps1:160]: I think that MSBuild (at least outside of MSFT employees) is significantly less well known, and much harder to find help online than powershell.  PowerShell also has PowerShell ISE that allows step by step debugging.

So, I think that converting powershell scripts into MSBuild is a step backwards.

Comment (jeffkl) [reply build.ps1:160]: In some cases I agree.  If this Powershell script ran locally and was used by us, Powershell could be the right answer.  I am not advocating that we take every Powershell script we have and replace it with MSBuild logic.

However, in this case the script is doing the following:

1. Assumes `bootstrap.proj` has been restored via `MSBuild /t:restore` or `dotnet restore`
2. Parses `bootstrap.proj` as XML so it knows the version of a package and forms a path to the NuGet package that was restored by `bootstrap.proj`.
3. Gets the output after _building_ `API.Test.csproj` and copies them to a specific location
4. Gets the output after _building_ `GenerateTestPackages.csproj` and copies them to a specific location
5. Copies files from the repository to a specific location

Most of that is relying on `MSBuild` to have restored packages and built projects.  Replacing the logic with an MSBuild project itself with standard `<ProjectReference />` items, a set of items, and a call to a `<Robocopy />` task doesn't seem like that big of a shift.

Also, this script runs on a remote machine most of the time and debugging what Powershell is doing remotely isn't great in my opinion.  The MSBuild binary logger is great for capturing what took place on a different machine.

File: build/BuildValidator.proj [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<Project Sdk="Microsoft.Build.NoTargets">
```
Comment (nkolev92) [inline build/BuildValidator.proj:1]: Can we name this `LocalizationValidator` or something like that/

I know the package is not named super well, but we can do better with the project.

File: build/loc.proj [hunk 3]
```diff
@@ -99,7 +106,4 @@
     <Copy SourceFiles="@(_EnglishBinaries)" DestinationFiles="@(_EnglishBinaries->'%(DestinationPath)')" />
     <Copy SourceFiles="@(_LocalizeFiles)" DestinationFiles="@(_LocalizeFiles->'%(DestinationPath)')" />
   </Target>
-
-  <Target Name="AfterBuild" DependsOnTargets="LocalizeNonProjectFiles;BatchLocalize"/>
-  <Import Project="$(MicroBuildDirectory)Microsoft.VisualStudioEng.MicroBuild.Core.targets" />
```
Comment (nkolev92) [inline build/loc.proj:104]: How are these targets included now?

File: setup/After.Microsoft.Common.targets [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<Project>
+  <PropertyGroup Condition="'$(MSBuildProjectExtension)' == '.vsmanproj'">
+    <OutputType />
```
Comment (nkolev92) [inline setup/After.Microsoft.Common.targets:3]: I'm imaging it's getting overwritten if we put it in the project file itself?

**Review by nkolev92 — COMMENTED** [review]
Love the overall intention. 

Would it be easier to have 1 PR to minimize the risk of the changes? 
All the changes are easy to follow individually, but there's a lot :D 

It would be helpful to summarize the changes in the PR description as well.

Have you verified that the outputs generated by this changes are equivalent before/after?

**Comment by ghost** [comment]
This PR has been automatically marked as stale because it has no activity for **7 days**. It will be closed if no further activity occurs **within another 7 days of this comment**. If it is closed, you may reopen it anytime when you're ready again, as long as you don't delete the branch.

**Pull request closed without merging**